oauth-common/src/main/java/io/strimzi/kafka/oauth/jsonpath/JsonPathFilterQuery.java

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright 2017-2019, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.jsonpath;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+
+/**
+ * This class implements parsing JsonPath filter query syntax inspired by:
+ *
+ *    https://github.com/json-path/JsonPath
+ *
+ * Given the following content of the JWT token:
+ * <pre>
+ *   {
+ *     "aud": ["uma_authorization", "kafka"],
+ *     "iss": "https://keycloak/token/",
+ *     "iat": 0,
+ *     "exp": 600,
+ *     "sub": "username",
+ *     "custom": "custom-value",
+ *     "roles": {
+ *       "client-roles": {
+ *         "kafka": ["kafka-user"]
+ *       }
+ *     },
+ *     "custom-level": 9
+ *   }
+ * </pre>
+ *
+ * Some examples of valid queries are:
+ *
+ * <pre>
+ *   "@.exp &lt; 1000"
+ *   "@.custom == 'custom-value'"
+ *   "@.custom == 'custom-value' and @.exp &gt; 1000"
+ *   "@.custom == 'custom-value' or @.exp &gt;= 1000"
```

Comment:
What's the precedence of `and` wrt `or`, and can you use brackets for force evaluation order?

Reply:
Evaluation is left to right, as soon as a false expression is followed by 'and' the evaluation short circuits to false. The brackets would be good yes. I left them out for now. Will see how hard it would be to add them.

Reply:
Implemented brackets and added short circuiting for true expressions followed by 'or'.